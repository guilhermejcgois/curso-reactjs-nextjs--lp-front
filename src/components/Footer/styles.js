import style, { css } from 'styled-components';

import { Container as SectionContainer } from '../SectionContainer/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = style.p`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
