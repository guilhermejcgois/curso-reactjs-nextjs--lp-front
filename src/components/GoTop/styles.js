import styled from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => `
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-contener: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
