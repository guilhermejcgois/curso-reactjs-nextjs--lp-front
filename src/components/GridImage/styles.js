import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
