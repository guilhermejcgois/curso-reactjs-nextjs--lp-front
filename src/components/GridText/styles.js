import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacing.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position absolute;
      font-size: ${theme.font.sizes.xhuge};
      top -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
