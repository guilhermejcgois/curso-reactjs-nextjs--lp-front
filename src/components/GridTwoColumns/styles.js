import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacing.xlarge};
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;
