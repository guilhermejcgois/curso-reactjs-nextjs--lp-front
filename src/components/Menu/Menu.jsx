import React, { useState } from 'react';
import P from 'prop-types';
import { Close as CloseIcon } from '@styled-icons/material-outlined/Close';
import { Menu as MenuIcon } from '@styled-icons/material-outlined/Menu';
import * as Styled from './styles';
import { LogoLink } from '../LogoLink';
import { NavLinks } from '../NavLinks';
import { SectionContainer } from '../SectionContainer';

export const Menu = ({ links = [], logoData }) => {
  const [visible, setVisible] = useState(false);
  return (
    <>
      <Styled.Button aria-label="Open/Close menu" visible={visible} onClick={() => setVisible(!visible)}>
        {visible ? <CloseIcon aria-label="Close menu" /> : <MenuIcon aria-label="Open menu" />}
      </Styled.Button>
      <Styled.Container visible={visible} onClick={() => setVisible(false)}>
        <SectionContainer>
          <Styled.MenuContainer>
            <LogoLink {...logoData} />
            <NavLinks links={links} />
          </Styled.MenuContainer>
        </SectionContainer>
      </Styled.Container>
    </>
  );
};

Menu.propTypes = {
  ...NavLinks.propTypes,
  logoData: P.shape(LogoLink.propTypes).isRequired,
};
