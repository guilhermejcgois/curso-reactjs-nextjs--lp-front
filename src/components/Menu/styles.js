import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ visible, theme }) => css`
    position: 0;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    width: 100%;
    height: 100%;
    border-bottom: ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;

    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && menuVisible()}

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      & ${Heading} {
        padding-bottom: ${theme.spacing.large};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacing.xxlarge};
    }
  `}
`;

export const Button = styled.button`
  ${({ visible, theme }) => css`
    display: none;
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    pointer-events: ${visible ? 'none' : 'all'};

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;
    }

    > svg {
      width: 2.5rem;
      height: 2.5rem;
    }
  `}
`;
