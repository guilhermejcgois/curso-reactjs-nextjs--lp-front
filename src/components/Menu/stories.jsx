import React from 'react';
import { Menu } from './Menu';

import linksMock from '../NavLinks/mock';

export default {
  title: 'Menu',
  component: Menu,
  args: {
    links: linksMock,
    logoData: {
      text: 'logo',
      link: '#target',
      srcImg: '',
    },
  },
};

export const Template = (args) => (
  <div>
    <Menu {...args} />
  </div>
);
