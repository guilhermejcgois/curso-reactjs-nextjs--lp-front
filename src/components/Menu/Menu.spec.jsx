import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';
import { Menu } from './Menu';
import linksMock from '../NavLinks/mock';

const logoData = {
  text: 'logo',
  link: '#target',
  srcImg: '',
};

describe('Menu', () => {
  it('should render Logo and Main Menu Nav', () => {
    const { container } = renderTheme(<Menu links={linksMock} logoData={logoData}></Menu>);
    expect(screen.getByRole('heading', { name: 'logo' })).toBeInTheDocument();
    expect(screen.getByRole('navigation', { name: 'Main menu' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render menu mobile and button for open and close the menu', () => {
    renderTheme(<Menu links={linksMock} logoData={logoData}></Menu>);

    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    expect(button).toHaveStyleRule('display', 'none');
    expect(button).toHaveStyleRule('display', 'flex', {
      media: theme.media.lteMedium,
    });
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.lteMedium,
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    fireEvent.click(button);
    expect(menuContainer).toHaveStyleRule('opacity', '1', {
      media: theme.media.lteMedium,
    });
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();

    fireEvent.click(menuContainer);
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.lteMedium,
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should not render links', () => {
    const { container } = renderTheme(<Menu logoData={logoData} />);
    expect(screen.getByRole('navigation', { name: 'Main menu' }).firstChild).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
