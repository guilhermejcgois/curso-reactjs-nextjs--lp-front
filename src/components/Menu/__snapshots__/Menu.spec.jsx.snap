// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should not render links 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  position: 0;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  border-bottom: #dddddd;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 > .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: none;
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0a1128;
  color: #ffffff;
  border: none;
  pointer-events: all;
}

.c0 > svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  max-height: 3rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row-wrap;
  -ms-flex-flow: row-wrap;
  flex-flow: row-wrap;
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 > .c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    overflow-y: auto;
  }

  .c2 {
    padding-bottom: 3.2rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
        >
          <a
            class="c7"
            href="#target"
          >
            <span>
              logo
            </span>
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c8"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Menu should render Logo and Main Menu Nav 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c2 {
  position: 0;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  border-bottom: #dddddd;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 > .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: none;
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0a1128;
  color: #ffffff;
  border: none;
  pointer-events: all;
}

.c0 > svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  max-height: 3rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row-wrap;
  -ms-flex-flow: row-wrap;
  flex-flow: row-wrap;
}

.c9 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c9::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c9:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 > .c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    overflow-y: auto;
  }

  .c2 {
    padding-bottom: 3.2rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c8 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6"
        >
          <a
            class="c7"
            href="#target"
          >
            <span>
              logo
            </span>
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c8"
        >
          <a
            class="c9"
            href="#target1"
            role="link"
            target="_self"
          >
            link 1
          </a>
          <a
            class="c9"
            href="#target2"
            role="link"
            target="_self"
          >
            link 2
          </a>
          <a
            class="c9"
            href="#target3"
            role="link"
            target="_self"
          >
            link 3
          </a>
          <a
            class="c9"
            href="#target4"
            role="link"
            target="_self"
          >
            link 4
          </a>
          <a
            class="c9"
            href="#target5"
            role="link"
            target="_self"
          >
            link 5
          </a>
          <a
            class="c9"
            href="#target6"
            role="link"
            target="_self"
          >
            link 6
          </a>
          <a
            class="c9"
            href="#target7"
            role="link"
            target="_self"
          >
            link 7
          </a>
          <a
            class="c9"
            href="#target8"
            role="link"
            target="_self"
          >
            link 8
          </a>
          <a
            class="c9"
            href="#target9"
            role="link"
            target="_self"
          >
            link 9
          </a>
          <a
            class="c9"
            href="#target10"
            role="link"
            target="_self"
          >
            link 10
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
